sunpy/map/mapbase.py

```diff
@@ -1566,19 +1566,12 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
                           ((pad_y, pad_y), (pad_x, pad_x)),
                           mode='constant',
                           constant_values=(missing, missing))
-        new_meta['crpix1'] += pad_x
-        new_meta['crpix2'] += pad_y
 
         # All of the following pixel calculations use a pixel origin of 0
 
         pixel_array_center = (np.flipud(new_data.shape) - 1) / 2.0
 
-        # Create a temporary map so we can use it for the data to pixel calculation.
-        temp_map = self._new_instance(new_data, new_meta, self.plot_settings)
-
-        # Convert the axis of rotation from data coordinates to pixel coordinates
-        pixel_rotation_center = u.Quantity(temp_map.world_to_pixel(self.reference_coordinate)).value
-        del temp_map
+        pixel_rotation_center = u.Quantity(self.reference_pixel).value + [pad_x, pad_y]
```

Comment:
```suggestion
        pixel_rotation_center = u.Quantity(self.reference_pixel).to_value(u.pix) + [pad_x, pad_y]
```

Reply:
Conceptually, sure, but this is not arbitrary input; we control the property `.reference_pixel`, and the units will always be pixels.  As a similar situation, I am fine with a bare `.value` on `.rsun_meters` because the units are a promise.